packages/upload/test/adding-files.test.js

```diff
@@ -27,36 +27,36 @@ describe('file list', () => {
 
     it('should notify files property changes', () => {
       const spy = sinon.spy();
-      // when setting arrays Polymer mutation observer works differently
       upload.addEventListener('files-changed', spy);
       upload.files = files;
       expect(spy.calledOnce).to.be.true;
     });
   });
 
-  (isFirefox ? describe.skip : describe)('with add button', () => {
+  describe('with add button', () => {
+    let input;
+    let addFiles;
+    let clickSpy;
+
+    beforeEach(() => {
+      addFiles = upload.$.addFiles;
+      input = upload.$.fileInput;
+      // While the synthetic "Add Files" button click event is not trusted and
+      // it should generate a non-trusted click event on the hidden file input,
+      // at the time of writing Chrome and Firefox still open the file dialog.
+      // Use stub calling `preventDefault` to prevent dialog from opening.
+      clickSpy = sinon.stub().callsFake((e) => e.preventDefault());
+      input.addEventListener('click', clickSpy);
+    });
+
     it('should open file dialog by click', () => {
-      // This test checks if the 'click' event is synchronously dispatched
-      // on the hidden file input when user clicks Add Files button. The
-      // file dialog is actually not getting opened during testing.
-      //
-      // The mock Add Files button click event fired by this test is not
-      // trusted and hence it should generate a non-trusted click event on
-      // the hidden file input. For security reasons non-IE browsers don't
-      // open file dialog when file input click is non-trusted.
-      const clickSpy = sinon.spy();
-      upload.$.fileInput.addEventListener('click', clickSpy);
-
-      upload.$.addFiles.dispatchEvent(new MouseEvent('click'));
+      addFiles.dispatchEvent(new MouseEvent('click'));
```

Comment:
**nit:** The `click()` helper of the `@vaadin/testing-helpers` package could be used here instead.

Reply:
Done.